### ISSUE TITLE
hi, when will the code be released ?

### ISSUE THREAD

**Issue by hanruijiang** [krennic999/STAR#8]

**Comment by krennic999** [comment]
emmmm... currently not sure, we are working on it

**Comment by waltstephen** [comment]
Waiting  for months XD

**Comment by krennic999** [comment]
errr....sorry...
We have tried for improving the VQVAE performance, but some problems occurred...
And we promise to make the weights and code public at the appropriate time...

**Comment by Owen718** [comment]
Could you please describe the issue you are encountering?

**Comment by krennic999** [comment]
I tried to create a multi-scale version based on current open-source f16 VQVAEs, but currently my VQVAE still has a certain performance gap compared to the version they released. I found it difficult to balance their quantizer, and after full-finetune for several epochs, the VQ loss will explode. I tried to modify the VQ loss and it worked for now, but the reconstruction quality is still not ideal; I think their official VQVAE may include a multi-stage training process.

In addition, I tried to figure out the role of each part in their VQVAE. The composition of their quantifier is very strange, and reconstruction seems to be composed not only of lookup from the codebook, but also of convolutions and decoders after each scale. Moreover, if the quantifier is removed, the reconstructed image using only the encoder and decoder will be very strange; But when I tried to solve these problems, I couldn't get a better VQVAE than their official release.

I' ll be glad if you are interested or can provide any suggestions🥹...
